### DISCUSSION TITLE
warning: wine cmd.exe /c echo '%AppData%' returned empty string, error message ""

### DISCUSSION THREAD

**Discussion by MohamedAliClai** [LSPosed/MagiskOnWSALocal#211] [Q&A] [answered]
Hi
i followed the instructions and when I run `scripts/run.sh` it give me some errors 

![image](https://user-images.githubusercontent.com/101479148/197389615-ed1b0121-3fe5-45d4-a0e0-79bd1e878dd8.png)

how can I fix that ?
note: I'm not a developer or even advanced, so please give step-by-step answer.

Thanks

**Comment by Howard20181** [comment, marked as answer]
Run the script under a non-root user

**Reply by MohamedAliClai** [reply]
Thanks for your reply
it still gives me the same error 

![image](https://user-images.githubusercontent.com/101479148/197390998-251f5fc9-1fe7-4d5c-aea6-40d2350ceca1.png)

**Reply by Howard20181** [reply]
Uninstall wine
Remove `~/.wine`
Re-run the script

**Reply by MohamedAliClai** [reply]
on root
-bash: /root/.wine: No such file or directory
on non root
-bash: /home/brink2/.wine: No such file or directory

**Reply by Howard20181** [reply]
If that folder does not exist it may be that wine failed to be configured during installation, uninstall it and reinstall it.

**Reply by MohamedAliClai** [reply]
how to uninstall
i don't even know file path or location
Is that command to uninstall it ?

**Reply by Howard20181** [reply]
`sudo apt remove wine && sudo apt autoremove`

**Reply by MohamedAliClai** [reply]
it didn't work after uninstalling and reinstall the wine
I just uninstalled then reinstalled Ubuntu from the Microsoft store and all works fine.
Thank you for your support and patience
